morphct/mobility_kmc.py

```diff
@@ -0,0 +1,694 @@
+    mol_id_dict : dict, default None
+        A dictionary that maps the chromophore index to the molecule index.
+
+        >>> mol_id_dict[chromophore_index]
+        ... molecule_index
+
+        Used to determine whether a hop is inter- or intra- molecular.
+    use_avg_hoprates : bool, default False
+       Whether to use the average hop rates instead of calculating each hop.
+    avg_intra_rate : float, default None
+        The average intramolecular hop rate in inverse seconds.
+    avg_inter_rate : float, default None
+        The average intermolecular hop rate in inverse seconds.
+    use_koopmans : bool, default False
+        Whether to use Koopman's approximation.
+    boltz : bool, default False
+        Whether to use a Boltzmann energy penalty
+    use_vrh : bool, default False
+        Whether to use variable-range hopping.
+    hopping_prefactor : float, default 1.0
+        A prefactor to the rate equation.
+
+    Attributes
+    ----------
+    id : int
+        The index of this carrier.
+    image : numpy.ndarray
+        The image of this carrier. Used to account for the periodic boundary.
+    initial_chromo : Chromophore
+        The chromophore on which the carrier was initialized.
+    current_chromo : Chromophore
+        The chromophore on which the carrier resides currently.
+    lambda_ij : float
+        The reorganization energy in eV.
+    hop_limit : int
+        The maximum number of hops.
+    temp : float
+        The temperature in Kelvin.
+    lifetime : float
+        The carrier lifetime in seconds.
+    current_time : float
+        The current time in the simulation. (Starts at 0.0.)
+    hole_history : scipy.sparse.lil_matrix, shape (n x n), dtype int
+        A sparse matrix to track which chromophores a hole has occupied.
+    electron_history : scipy.sparse.lil_matrix, shape (n x n), dtype int
+        A sparse matrix to track which chromophores an electron has occupied.
+    c_type : str
+        The carrier type, "electron" or "hole".
+    n_hops : int
+        The number of hops the carrier has performed.
+    box : numpy.ndarray
+        The lengths of the box vectors of the simulation box.
+    displacement : float
+        The net displacement of the carrier in Angstroms.
+    mol_id_dict : dict
+        A dictionary that maps the chromophore index to the molecule index.
+    use_avg_hoprates : bool
+       Whether to use the average hop rates instead of calculating each hop.
+    avg_intra_rate : float
+        The average intramolecular hop rate in inverse seconds.
+    avg_inter_rate : float
+        The average intermolecular hop rate in inverse seconds.
+    use_koopmans : bool
+        Whether to use Koopman's approximation.
+    boltz : bool
+        Whether to use a Boltzmann energy penalty
+    use_vrh : bool
+        Whether to use variable-range hopping.
+    vrh_delocalization : float
+        The variable-range hopping rate (?) TODO EJ
```

Comment:
This is a parameter for adding a penalty to the transfer integral (scaling it down) if the distance between two chromophores is beyond some threshold.  So the docstring could be:

Hopping rates are scaled by exp(-r/vrh_delocalization) if use_vrh==True.